test/ziggurat/init_test.clj

```diff
@@ -3,26 +3,36 @@
             [mount.core :refer [defstate] :as mount]
             [ziggurat.config :as config]
             [ziggurat.init :as init]
-            [ziggurat.messaging.rabbitmq-wrapper :as rmqw]
             [ziggurat.messaging.consumer :as messaging-consumer]
             [ziggurat.messaging.producer :as messaging-producer]
             [ziggurat.streams :as streams :refer [stream]]
             [ziggurat.server.test-utils :as tu]
-            [ziggurat.tracer :as tracer])
+            [ziggurat.tracer :as tracer]
+            [ziggurat.messaging.messaging :as messaging])
   (:import (io.opentracing.mock MockTracer)))
 
+(def valid-modes-count 4)
+
+(defn- exp [x n]
+  (loop [acc 1 n n]
+    (if (zero? n) acc
+                  (recur (* x acc) (dec n)))))
```

Comment:
we know that n will not go above a limit, so why not just use plain recursion? There's no need for tail recursion here